pkg/porter/credentials.go

```diff
@@ -106,22 +107,91 @@ func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
 	if err != nil {
 		return err
 	}
-	bundle, err := p.CNAB.LoadBundle(opts.CNABFile)
 
+	_, err = p.generateNewCredentialSet(opts.CNABFile, opts.Name, opts.Silent, opts.DryRun)
+	return err
+}
+
+func (p *Porter) chooseCredentialSet(CNABFile string) (string, error) {
+	var credSetNames []string
+	credSets, err := p.Credentials.ReadAll()
 	if err != nil {
-		return err
+		return "", errors.Wrap(err, "failed to read exisiting credential sets")
+	}
+
+	shouldGenerateCred := false
+	var selectedOption string
+	if len(credSets) > 0 {
+		for _, credSet := range credSets {
+			credSetNames = append(credSetNames, credSet.Name)
+		}
+
+		selectCredPrompt := &survey.Select{
+			Message: "Choose a set of credentials to use while installing this bundle",
+			Options: append(credSetNames, generateCredCode, quitGenerateCode),
+			Default: credSetNames[0],
+		}
+		survey.AskOne(selectCredPrompt, &selectedOption, nil)
+
+		switch selectedOption {
+		case generateCredCode:
+			shouldGenerateCred = true
+		case quitGenerateCode:
+			return "", errors.New("Credentials are mandatory to install this bundle")
+		default:
+			return selectedOption, nil
+		}
+	} else {
+		shouldGenerateCredPrompt := &survey.Confirm{
+			Message: "No credential identifier given. Generate one ?",
+		}
+		survey.AskOne(shouldGenerateCredPrompt, &shouldGenerateCred, nil)
+	}
+
+	if !shouldGenerateCred {
+		return "", errors.New("Credentials are mandatory to install this bundle")
```

Comment:
```suggestion
		return "", errors.New("Credentials are mandatory to install this bundle but none were provided with the `--cred` flag")
```

I think it would help to tell them what to do to fix the command when they run it again. Both here and above.

Reply:
👍